feature/manager_test.go

```diff
@@ -0,0 +1,102 @@
+package feature_test
+
+import (
+	"reflect"
+	"testing"
+
+	"github.com/lightningnetwork/lnd/feature"
+	"github.com/lightningnetwork/lnd/lnwire"
+)
+
+type managerTest struct {
+	name string
+	cfg  feature.Config
+}
+
+var managerTests = []managerTest{
+	{
+		name: "default",
+		cfg:  feature.Config{},
+	},
+	{
+		name: "no tlv",
+		cfg: feature.Config{
+			NoTLVOnion: true,
+		},
+	},
+	{
+		name: "no static remote key",
+		cfg: feature.Config{
+			NoStaticRemoteKey: true,
+		},
+	},
+	{
+		name: "no tlv or static remote key",
+		cfg: feature.Config{
+			NoTLVOnion:        true,
+			NoStaticRemoteKey: true,
+		},
+	},
+}
+
+// TestManager asserts basic initialazation and operation of a feature manager,
+// including that the proper features are removed in response to config changes.
+func TestManager(t *testing.T) {
+	for _, test := range managerTests {
+		test := test
+		t.Run(test.name, func(t *testing.T) {
+			testManager(t, test)
+		})
+	}
+}
+
+func testManager(t *testing.T, test managerTest) {
+	m, err := feature.NewManager(test.cfg)
```

Comment:
Should `defaultSets` be a parameter of `NewManager`, to isolate the test from the current feature set?

Reply:
done